### PR TITLE
BUG: stats: Fix broadcasting in wrapcauchy.cdf

### LINKED ISSUE #13791: BUG: stats: wrapcauchy.cdf does not broadcast the shape parameter correctly.
When an array is given as the shape parameter to `wrapcauchy.cdf`, it appears that only the first value in the array is used in the calculations.  For example,

```
In [3]: c = np.array([0.03, 0.25, 0.5])                                                                                                             

In [4]: wrapcauchy.cdf(2.0, c)                                                                                                                      
Out[4]: array([0.3268839, 0.3268839, 0.3268839])
```
All the values are the same in that output, and they correspond to the CDF when the shape is 0.03.

Here are the correct values:
```

In [5]: [wrapcauchy.cdf(2, t) for t in c]                                                                                                           
Out[5]: [0.32688389546873015, 0.3829474595854628, 0.43288446668740754]
```

SciPy and NumPy versions:
```
In [7]: scipy.__version__                                                                                                                           
Out[7]: '1.7.0.dev0+08390dc'

In [8]: np.__version__                                                                                                                              
Out[8]: '1.19.5'
```

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Also added a few tests for specific values of the CDF.

Closes gh-13791.

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
Looks like CircleCI is having problems in recent PRs.  The changes here should have no effect on the docs.

CircleCI doesn't show the errors if you don't login in CircleCI, and I'm still resistant to giving yet another service write access to my github repos, so I'm not sure what is going wrong.  I can build the docs locally with only the usual warnings.

**Comment by tylerjereddy** [comment]
> CircleCI doesn't show the errors if you don't login to CircleCI, and I'm still resistant to giving yet another service write access to my github repos, so I'm not sure what is going wrong.

CircleCI knows all my secrets so I've opened the issue here: gh-13793

**Comment by mdhaber** [comment]
I think the CircleCI issue is fixed but you'd need to merge master.

**Comment by mdhaber** [comment]
I could probably study it and figure it out, but how did you save so many lines? What was the calculation doing that was unnecessary?

**Comment by WarrenWeckesser** [comment]
A fix with fewer lines changed is to just add the line

        x, c = np.broadcast_arrays(x, c)

immediately after `def _cdf(self, x, c):`.  But `_lazywhere` already does (more or less) what this function's code was doing, so we might as well continue following the `_lazywhere` pattern.

I did a couple performance comparisons, and the `_lazywhere` version was either very close to or better than the existing version.  For example, with `x = np.linspace(0, 2*np.pi, 500)` and `c = np.array([[0.03], [0.25], [0.75]])`, the call `wrapcauchy.cdf(x, c)` took 349 µs with the `_lazywhere` version, while the original version (with the broadcasting fix) took 373 µs.

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -8483,24 +8485,17 @@ def _pdf(self, x, c):
         return (1.0-c*c)/(2*np.pi*(1+c*c-2*c*np.cos(x)))
 
     def _cdf(self, x, c):
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:8487]:
```python3
from sympy import symbols, atan, tan, pi, cos, diff, simplify
c, x, z = symbols('c, x, z')
pdf = (1-c**2) / (2*pi*(1+c**2-2*c*cos(x)))
cr = (1 + c)/(1 - c)
cdf = 1/pi*atan(cr*tan(x/2))
pdf2 = diff(cdf, x)
print(simplify(pdf - pdf2))
```

**Comment by mdhaber** [comment]
Checked carefully. Looks good. Thanks @WarrenWeckesser.

**Pull request merged**